openlcs/libs/kojiconnector.py

```diff
@@ -536,3 +536,40 @@ def get_module_components(self, module_nvr):
             'RPMMOD': module_component
         }
         return components
+
+    def get_task_id(self, build):
+        """
+        Return build task_id.
+        """
+        extra = build.get('extra', None)
+        task_id = extra.get('container_koji_task_id') if extra else None
+        return task_id if task_id else None
+
+    def get_task_repository(self, task_id):
+        """
+        Return the task repository.
+        """
+        task_result = self._service.getTaskResult(task_id, raise_fault=True)
+        if task_result.get('repositories'):
+            repositories = task_result.get('repositories')
+        # The repository is a list. Generally, it only has the x86_64
+        # repository. But if the source container has different arch image
+        # archives, the repository list length is more than one. Here, I would
+        # like to fetch the first repository. If there are some differences
+        # between different arch of the source, pls FIX me.
+        return repositories[0] if repositories[0] else None
```

Comment:
you're assuming `repositories` always exists, which is untrue. see line above.

Reply:
Updated.